components/board/genericlinux/gpio.go

```diff
@@ -210,38 +210,6 @@ func (pin *gpioPin) startSoftwarePWM() error {
 	return nil
 }
 
-// accurateSleep is intended to be a replacement for utils.SelectContextOrWait which wakes up
-// closer to when it's supposed to. We return whether the context is still valid (not yet
-// cancelled).
-func accurateSleep(ctx context.Context, duration time.Duration) bool {
-	// If we use utils.SelectContextOrWait(), we will wake up sometime after when we're supposed
-	// to, which can be hundreds of microseconds later (because the process scheduler in the OS only
-	// schedules things every millisecond or two). For use cases like a web server responding to a
-	// query, that's fine. but when outputting a PWM signal, hundreds of microseconds can be a big
-	// deal. To avoid this, we sleep for less time than we're supposed to, and then busy-wait until
-	// the right time. Inspiration for this approach was taken from
-	// https://blog.bearcats.nl/accurate-sleep-function/
-	// On a raspberry pi 4, naively calling utils.SelectContextOrWait tended to have an error of
-	// about 140-300 microseconds, while this version had an error of 0.3-0.6 microseconds.
-	startTime := time.Now()
-	maxBusyWaitTime := 1500 * time.Microsecond
-	if duration > maxBusyWaitTime {
-		shorterDuration := duration - maxBusyWaitTime
-		if !utils.SelectContextOrWait(ctx, shorterDuration) {
-			return false
-		}
-	}
-
-	for time.Since(startTime) < duration {
-		select {
-		case <-ctx.Done():
-			return false
-		default: // Busy-wait some more
-		}
```

Comment:
You don't need the select here

try:
```
if ctx.Err != nil {
   return false
}
```

Reply:
Needs parentheses in `ctx.Err()`, but otherwise this is a good call! If we keep this code in, I should change that. James had made a similar comment when this was added in, but your version is clearer.